program/tests/close_lookup_table_ix.rs

```diff
@@ -120,34 +139,60 @@ async fn test_close_lookup_table_deactivated_in_current_slot() {
 
 #[tokio::test]
 async fn test_close_lookup_table_recently_deactivated() {
+    // Try to close a lookup table that was deactivated in a previous slot,
+    // but the cooldown period hasn't expired yet.
+    // This should fail because the table must be deactivated in a previous
+    // slot and the cooldown period must expire before it can be closed.
     let mut context = setup_test_context().await;
 
     let authority_keypair = Keypair::new();
-    let initialized_table = {
-        let mut table = new_address_lookup_table(Some(authority_keypair.pubkey()), 0);
-        table.meta.deactivation_slot = 0;
-        table
-    };
-    let lookup_table_address = Pubkey::new_unique();
-    add_lookup_table_account(&mut context, lookup_table_address, initialized_table).await;
 
-    let ix = close_lookup_table(
-        lookup_table_address,
-        authority_keypair.pubkey(),
-        context.payer.pubkey(),
-    );
+    // [Core BPF]: The original builtin implementation was relying on the fact
+    // that the `SlotHashes` sysvar is initialized to have an entry for slot 0.
+    // Program-Test does this to provide a more realistic test environment.
+    // That means this test was running with the `Clock` current slot at 1.
+    // In this implementation, we adapt the deactivation slot as well as the
+    // current slot into tweakable test case values.
+    for (deactivation_slot, current_slot) in [
+        (0, 1),
+        (0, 40),  // Arbitrary number within cooldown (ie. 41 slot hashes 0..40).
+        (0, 511), // At the very edge of cooldown (ie. 512 slot hashes 0..511).
```

Comment:
This test is pretty great, seems like you could fold in the previous test into this with a test case of `(0, 0)`, right?

Reply:
Done!